### DISCUSSION TITLE
Content in enumerate sometimes escapes bounding box

### DISCUSSION THREAD

**Discussion by RafailFridman** [typst/typst#1549] [Q&A] [answered]
I'm trying to create a template for my HW in typst:
```
#let tasks_counter = counter("tasks_counter")
#let task(input_text, solution_space: 10pt) = {
  tasks_counter.step()
  let problem_text = text(13pt, weight: "bold")[Problem #tasks_counter.display().]
  let task_text = strong(underline("\nTask:\n"))
  let solution_text = underline("Solution:\n")
  let full_text = problem_text + task_text + input_text
  layout(page_size => style(styles => grid(
  columns: 2,
  rows: measure(box(width: page_size.width, full_text), styles).height,
  line(
    angle: 90deg,
    length: 100%,
    start: (2%, 0%),
    end: (2%, 100%),
    stroke: (thickness: 2pt)
  ),
 full_text,
)))
v(solution_space, weak: false)
solution_text
}
```

Then, when I use it:
```
#task[
  #lorem(40)
  + #lorem(20)
]
```
It usually works:
![image](https://github.com/typst/typst/assets/22198039/965d359a-033b-4df2-9551-e6b31ddb287f)

But if there is a hanging line in enumerate, the contents escape the vertical black line on the right:
```
#task[
  + #lorem(31)
  + #lorem(31)
]
```
![image](https://github.com/typst/typst/assets/22198039/fd87bbaf-b5f3-44d1-8c42-8c92f10c5c77)

Is it my problem of not correctly defining the bounding box or is it a known problem?

**Comment by RafailFridman** [comment]
I noticed that when I change the part
```
line(
    angle: 90deg,
    length: 100%,
    start: (2%, 0%),
    end: (2%, 100%),
    stroke: (thickness: 2pt)
  )
```
 to be 
```
line(
    angle: 90deg,
    length: 100%,
    start: (0%, 0%),
    end: (0%, 100%),
    stroke: (thickness: 2pt)
  )
```
(notice that now 2% became 0%). Then it looks ugly, but seems to be behaving correctly:
```
#task[
  + #lorem(46)
  + #lorem(46)
]
```

![image](https://github.com/typst/typst/assets/22198039/c733eeaa-c8ae-419c-b6c2-c13d94734d10)

**Comment by Enivex** [comment]
I didn't read this in full detail, but I'm pretty sure your problem is here

`measure(box(width: page_size.width, full_text), styles)`

You're measuring with respect to the entire page width, but then give it only 98% of it after.

Edit: As an aside, a much easier way to achieve this without needing measure at all is just adding stroke on the left side of the text block, with appropriate inset or outset there as well.

**Reply by RafailFridman** [reply]
Thanks!
Could you please point me to the stroke thing you described? 
I see in the docs it is a parameter of many functions.

**Reply by Enivex** [reply, marked as answer]
Try using block(stroke: (left: 1pt),inset: (left: .5em), body) for instance. If you want more pointers I'd recommend checking out the discord. It's a bit easier to explain things there because of the bot.

**Reply by RafailFridman** [reply]
Wow, so much easier! Thanks again
